src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/prison/PrisonerSearchApiClient.kt

```diff
@@ -61,4 +63,30 @@ class PrisonerSearchApiClient(@Qualifier("oauthPrisonerSearchClient") val prison
       .block()?.toPage() ?: Page.empty()
     return result
   }
+
+  fun getAllByReleaseDate(from: LocalDate, to: LocalDate, pageSize: Int = PAGE_SIZE): List<PrisonerSearchPrisoner> {
+    val result = mutableListOf<PrisonerSearchPrisoner>()
+    var pageNumber = 0
+
+    while (pageNumber >= 0) {
```

Comment:
iterates over all records coming up to records.. considered mapping to a simpler representation to reduce amount of in-memory state.